### PR TITLE
MINOR: improve QueryableStateIntegrationTest

### PR DESCRIPTION
**Pull request by mjsax**
Fix test `Comparators` plus Java8 cleanup

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java [hunk 1]
```diff
@@ -194,23 +192,8 @@ public void before() throws Exception {
...
-        stringLongComparator = new Comparator<KeyValue<String, Long>>() {
-
-            @Override
-            public int compare(final KeyValue<String, Long> o1,
-                               final KeyValue<String, Long> o2) {
-                return o1.key.compareTo(o2.key);
-            }
-        };
+        stringComparator = Comparator.comparing((KeyValue<String, String> o) -> o.key).thenComparing(o -> o.value);
+        stringLongComparator = Comparator.comparing((KeyValue<String, Long> o) -> o.key).thenComparingLong(o -> o.value);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java:196]: This is the actual fix.

**Comment by mjsax** [comment]
Call for review @guozhangwang @bbejeck @vvcephei 

This should be cherry-picked into `2.1` and `2.0`, too. For older branches, there is #5988

**Comment by guozhangwang** [comment]
Build timed out. Local run passed.

**Review by guozhangwang — APPROVED** [review]

**Comment by guozhangwang** [comment]
Cherry-picked to 2.1 and 2.0 (2.0 has lots of conflicts, so I just pushed the one-liner of the actual fix).

**Comment by mjsax** [comment]
> 2.0 has lots of conflicts, so I just pushed the one-liner of the actual fi

Cool. That's good enough. Thank you!

**Pull request merged**